
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Data Pre-processing &mdash; phyluce v0.1.0 documentation</title>
    <link rel="stylesheet" href="_static/flasky.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '',
        VERSION:     '0.1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="phyluce v0.1.0 documentation" href="index.html" />
    <link rel="next" title="Two roads diverged in a yellow wood" href="next-steps.html" />
    <link rel="prev" title="Study design" href="study-design.html" />
   
  
  <link media="only screen and (max-device-width: 480px)" href="_static/small_flask.css" type= "text/css" rel="stylesheet" />

  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="next-steps.html" title="Two roads diverged in a yellow wood"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="study-design.html" title="Study design"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">phyluce v0.1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="data-pre-processing">
<h1>Data Pre-processing<a class="headerlink" href="#data-pre-processing" title="Permalink to this headline">¶</a></h1>
<div class="section" id="demultiplexing">
<h2>Demultiplexing<a class="headerlink" href="#demultiplexing" title="Permalink to this headline">¶</a></h2>
<p>Because we are reducing genomes, and beceause we generally reduce genomes to a
fraction of their original size, we can combine DNA libraries from many sources
into a single lane of HiSeq (or even MiSeq) sequencing.  Therefore, because we
are multiplexing these libraries prior to sequencing, we need to &#8220;demultiplex&#8221;
these libraries after they&#8217;re sequenced.</p>
<p>Demultiplexing is a tricky process, made tricker by the platform with which you
are working.  Thankfully, demultiplexing (or &#8220;demuxing&#8221;) Illumina reads is
rather simple. There are several options:</p>
<ul class="simple">
<li>the Illumina pipeline</li>
<li><a class="reference external" href="http://hannonlab.cshl.edu/fastx_toolkit/">FASTX-toolkit</a></li>
</ul>
<p>I have also written <a class="reference external" href="https://github.com/faircloth-lab/splitaake/">splitaake</a>, which is a program to help demultiplex
Illumina reads.  <a class="reference external" href="https://github.com/faircloth-lab/splitaake/">splitaake</a> is particularly useful when you have very many
sequence tags (aka &#8220;barcodes&#8221;) to demultiplex, when your sequence tags are
long, or some combination of the two.</p>
<p>To demultiplex your data with <a class="reference external" href="https://github.com/faircloth-lab/splitaake/">splitaake</a>, you need a couple of things - first
you need to <a href="#id1"><span class="problematic" id="id2">`install splitaake`_</span></a>.  Second, you need to create a configuration
file (e.g. <tt class="docutils literal"><span class="pre">my-test-file.conf</span></tt>) for <a class="reference external" href="https://github.com/faircloth-lab/splitaake/">splitaake</a> that relates your sequence
tags to their samples.  Here  is an example:</p>
<div class="highlight-python"><pre>[L007]
#name:sequence-tag
BFIDT-000:AACCGAGTTA
BFIDT-001:AATACTTCCG
BFIDT-002:AATTAAGGCC
BFIDT-003:AAGCTTATCC
BFIDT-004:CTAACGATCG
BFIDT-005:CTACAACGGC</pre>
</div>
<p>Once all that&#8217;s done, you may want to start up <a class="reference external" href="https://github.com/faircloth-lab/splitaake/">splitaake</a> in a <a class="reference external" href="http://www.gnu.org/software/screen/">screen</a> or
<a class="reference external" href="http://tmux.sourceforge.net/">tmux</a> session by running:</p>
<div class="highlight-bash"><div class="highlight"><pre>python splitaake.py r1.fastq.gz r2.fastq.gz r3.fastq.gz <span class="se">\</span>
                my-test-file.conf --section L007
</pre></div>
</div>
<p>This can take a rather long time for a single lane of 150-190 million Illumina
reads - on the order of 15-18 hours for 50-60 sequence tags.  These are ways to
speed up this process, but it&#8217;s not always as easy as it seems (I/O becomes a
problem very rapidly).</p>
<p>When you&#8217;re done, regardless of the program that you used, you should have
something similar to a directory containing your reads data in fastq format:</p>
<div class="highlight-python"><pre>/Data
        L007/
                r1.fastq.gz
                r2.fastq.gz
                r3.fastq.gz
                dmux/
                        BFIDT-000.fastq.gz
                        BFIDT-001.fastq.gz
                        BFIDT-002.fastq.gz
                        BFIDT-003.fastq.gz
                        BFIDT-004.fastq.gz
                        BFIDT-005.fastq.gz</pre>
</div>
<p>In the above, the reads from the original files <tt class="docutils literal"><span class="pre">r1.fastq.gz</span></tt>,
<tt class="docutils literal"><span class="pre">r2.fastq.gz</span></tt>, <tt class="docutils literal"><span class="pre">r2.fastq.gz</span></tt>, which reside in <tt class="docutils literal"><span class="pre">/Data/L007</span></tt> have been split
into their component parts - with each name in the configuration file above
becoming its own <a class="reference external" href="http://www.gzip.org/">gzip</a> file of results.</p>
<p>Additionally, your resulting, demutliplexed files will be named however you
setup the configuration file (e.g. <tt class="docutils literal"><span class="pre">my-test-file.conf</span></tt>).  If you had created
that file as follows:</p>
<div class="highlight-python"><pre>[bins]
#name:sequence-tag
chicken:AACCGAGTTA
duck:AATACTTCCG
quail:AATTAAGGCC
Turkey:AAGCTTATCC</pre>
</div>
<p>Then your resulting data would be in <a class="reference external" href="http://www.gzip.org/">gzip</a> files named:</p>
<div class="highlight-python"><pre>/Data
        L007/
                r1.fastq.gz
                r2.fastq.gz
                r3.fastq.gz
                dmux/
                        chicken.fastq.gz
                        duck.fastq.gz
                        quail.fastq.gz
                        turkey.fastq.gz</pre>
</div>
</div>
<div class="section" id="adapter-and-quality-trimming">
<h2>Adapter- and quality- trimming<a class="headerlink" href="#adapter-and-quality-trimming" title="Permalink to this headline">¶</a></h2>
<p>Once you&#8217;ve demultiplexed your reads, you generally need to do some quality
control on what&#8217;s in each of your names fastq.gzip files.  The reasons for this
are several-fold.  Generally speaking, you&#8217;re dealing with the fact that the
ends of some reads tends to be of relatively low quality, and some reads have
ambiguous bases, which we generally want to avoid.  Finally, some reads, for
whatever reason, may still have some adapter sequence in the actual read - an
outcome that you may see referred to as <cite>vector contamination</cite> even though we&#8217;re
not using vectors.</p>
<p>Anyway, bottom-line is that we want to do a little trimming on the reads to
cut off low-quality ends and any left-over adapter that&#8217;s made it&#8217;s way into the
read.</p>
<p>As with demultiplexing, there are any number of ways of doing this.  Here are
some programs that do parts of what you need to do:</p>
<ul class="simple">
<li><a class="reference external" href="http://hannonlab.cshl.edu/fastx_toolkit/">FASTX-toolkit</a></li>
<li></li>
</ul>
<p>You may also be interested in doing a little pre-trimming visualization to get
an idea of how things are looking - always the best way to make decisions.  For
that, <a class="reference external" href="http://vincebuffalo.com/">Vince Buffalo</a> has a pretty bad-ass slice of code named <a class="reference external" href="https://github.com/vsbuffalo/qrqc">qrqc</a> (see
figure above).</p>
<p>Because there are several steps, I generally automate this process using some
code I wrote called <a class="reference external" href="https://github.com/faircloth-lab/illumiprocessor/">illumiprocessor</a> - it&#8217;s basically a bit of code that just
runs the component parts of this process that I want to run, outputting some
stats and other goodies that we&#8217;ll use later.</p>
<p>Additionally, <a class="reference external" href="https://github.com/faircloth-lab/illumiprocessor/">illumiprocessor</a> sets up a directory structure that we&#8217;ll use
during the rest of our analyses - so you&#8217;ll be ready to go once it&#8217;s done
running.  Please feel free to use whichever program/script that you prefer. The
final directory structure you need to make things run smoothly once QC is
finished looks like:</p>
<div class="highlight-python"><pre>chicken/
        chicken.fastq.gz
        chicken.fastq-trimmed.gz
        chicken.stats.txt
duck/
        duck.fastq.gz
        duck.fastq-trimmed.gz
        duck.stats.txt
quail/
        quail.fastq.gz
        quail.fastq-trimmed.gz
        quail.stats.txt</pre>
</div>
<p>If you setup a configuration file for <a class="reference external" href="https://github.com/faircloth-lab/splitaake/">splitaake</a> above, then we&#8217;re just going
to add some bits to it.  If you don&#8217;t have a config file, then we need to set
one up.  This is not terribly bad:</p>
<div class="highlight-python"><pre>[io]
input:
output:

[steps]
trim:True
clip:True
ns:True
stats:True

[jobs]
cores:Auto

[bins]
#name:sequence-tag
BFIDT-000:AACCGAGTTA
BFIDT-001:AATACTTCCG
BFIDT-002:AATTAAGGCC
BFIDT-003:AAGCTTATCC
BFIDT-004:CTAACGATCG
BFIDT-005:CTACAACGGC</pre>
</div>
<p>One that&#8217;s done, we&#8217;re ready to rock.  The analysis will run on multiple cores
if you have them and <tt class="docutils literal"><span class="pre">cores</span></tt> above is set to AUTO.  If <tt class="docutils literal"><span class="pre">cores</span></tt> is set to an
integer value, the code will use that many cores to analyze your data.  There
are more complex options available, so check the documentation for
<a class="reference external" href="https://github.com/faircloth-lab/illumiprocessor/">illumiprocessor</a> if you want to get fancy.</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Data Pre-processing</a><ul>
<li><a class="reference internal" href="#demultiplexing">Demultiplexing</a></li>
<li><a class="reference internal" href="#adapter-and-quality-trimming">Adapter- and quality- trimming</a></li>
</ul>
</li>
</ul>
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="study-design.html" title="previous chapter">Study design</a></li>
      <li>Next: <a href="next-steps.html" title="next chapter">Two roads diverged in a yellow wood</a></li>
  </ul></li>
</ul>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/pre-processing.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy; Copyright 2012, Brant C. Faircloth.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a>.
    </div>
  </body>
</html>