
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Data Pre-processing &mdash; phyluce v0.1.0 documentation</title>
    <link rel="stylesheet" href="_static/flasky.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '',
        VERSION:     '0.1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="phyluce v0.1.0 documentation" href="index.html" />
    <link rel="next" title="&lt;no title&gt;" href="uce-processing.html" />
    <link rel="prev" title="Study design" href="study-design.html" />
   
  
  <link media="only screen and (max-device-width: 480px)" href="_static/small_flask.css" type= "text/css" rel="stylesheet" />

  </head>
  <body>
    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="uce-processing.html" title="&lt;no title&gt;"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="study-design.html" title="Study design"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">phyluce v0.1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="data-pre-processing">
<h1>Data Pre-processing<a class="headerlink" href="#data-pre-processing" title="Permalink to this headline">¶</a></h1>
<div class="section" id="demultiplexing">
<h2>Demultiplexing<a class="headerlink" href="#demultiplexing" title="Permalink to this headline">¶</a></h2>
<p>Because we are reducing genomes, and beceause we generally reduce genomes to a
fraction of their original size, we can combine DNA libraries from many sources
into a single lane of HiSeq (or even MiSeq) sequencing.  Therefore, because we
are multiplexing these libraries prior to sequencing, we need to &#8220;demultiplex&#8221;
these libraries after they&#8217;re sequenced.</p>
<p>Demultiplexing is a tricky process, made tricker by the platform with which you
are working.  Thankfully, demultiplexing (or &#8220;demuxing&#8221;) Illumina reads is
rather simple. There are several options:</p>
<ul class="simple">
<li>the Illumina pipeline</li>
<li><a href="#id1"><span class="problematic" id="id2">`fastx-tools &lt;&gt;`_</span></a></li>
</ul>
<p>I have also written <a class="reference external" href="https://github.com/faircloth-lab/splitaake/">splitaake</a>, which is a program to help demultiplex
Illumina reads.  <a class="reference external" href="https://github.com/faircloth-lab/splitaake/">splitaake</a> is particularly useful when you have very many
sequence tags (aka &#8220;barcodes&#8221;) to demultiplex, when your sequence tags are
long, or some combination of the two.</p>
<p>To demultiplex your data with <a class="reference external" href="https://github.com/faircloth-lab/splitaake/">splitaake</a>, you need a couple of things - first
you need to <a href="#id3"><span class="problematic" id="id4">`install splitaake &lt;&gt;`_</span></a>.  Second, you need to create a configuration
file (e.g. my-test-file.conf) for <a class="reference external" href="https://github.com/faircloth-lab/splitaake/">splitaake</a> that relates your sequence tags
to their samples.  Here  is an example:</p>
<div class="highlight-python"><pre>[L007]
#name:sequence-tag
BFIDT-000:AACCGAGTTA
BFIDT-001:AATACTTCCG
BFIDT-002:AATTAAGGCC
BFIDT-003:AAGCTTATCC
BFIDT-004:CTAACGATCG
BFIDT-005:CTACAACGGC</pre>
</div>
<p>Once all that&#8217;s done, you may want to start up <a class="reference external" href="https://github.com/faircloth-lab/splitaake/">splitaake</a> in a <a class="reference external" href="http://www.gnu.org/software/screen/">screen</a> or
<a class="reference external" href="http://tmux.sourceforge.net/">tmux</a> session by running:</p>
<div class="highlight-bash"><div class="highlight"><pre>python dmux.py r1.fastq.gz r2.fastq.gz r3.fastq.gz <span class="se">\</span>
                my-test-file.conf --section L007
</pre></div>
</div>
<p>This can take a rather long time for a single lane of 150-190 million Illumina
reads - on the order of 15-18 hours for 50-60 sequence tags.  These are ways to
speed up this process, but it&#8217;s not always as easy as it seems (I/O becomes a
problem very rapidly).</p>
<p>When you&#8217;re done, regardless of the program that you used, you should have
something similar to a directory containing your reads data in fastq format:</p>
<div class="highlight-python"><pre>/Data
        L007/
                r1.fastq.gz
                r2.fastq.gz
                r3.fastq.gz
                dmux/
                        BFIDT-000.fastq.gz
                        BFIDT-001.fastq.gz
                        BFIDT-002.fastq.gz
                        BFIDT-003.fastq.gz
                        BFIDT-004.fastq.gz
                        BFIDT-005.fastq.gz</pre>
</div>
<p>In the above, the reasd from the original files <cite>r1.fastq.gz</cite>, <cite>r2.fastq.gz</cite>,
<cite>r2.fastq.gz</cite> have been split into their component parts - with each name in the
configuration file above becoming its own gzip file of results.</p>
</div>
<div class="section" id="adapter-and-quality-trimming">
<h2>Adapter- and quality- trimming<a class="headerlink" href="#adapter-and-quality-trimming" title="Permalink to this headline">¶</a></h2>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Data Pre-processing</a><ul>
<li><a class="reference internal" href="#demultiplexing">Demultiplexing</a></li>
<li><a class="reference internal" href="#adapter-and-quality-trimming">Adapter- and quality- trimming</a></li>
</ul>
</li>
</ul>
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="study-design.html" title="previous chapter">Study design</a></li>
      <li>Next: <a href="uce-processing.html" title="next chapter">&lt;no title&gt;</a></li>
  </ul></li>
</ul>
  <h3>This Page</h3>
  <ul class="this-page-menu">
    <li><a href="_sources/pre-processing.txt"
           rel="nofollow">Show Source</a></li>
  </ul>
<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy; Copyright 2012, Brant C. Faircloth.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a>.
    </div>
  </body>
</html>